sdk/enoki/src/react.tsx

```diff
@@ -67,3 +74,45 @@ export function useAuthCallback() {
 
 	return { handled, state };
 }
+
+export const EnokiWalletContext = createContext<{
+	wallets: ReturnType<typeof registerEnokiWallets>['wallets'];
+	client: SuiClient;
+} | null>(null);
+
+export function EnokiWalletProvider({
+	children,
+	config,
+	useSuiClientContext,
+}: {
+	config: Omit<Parameters<typeof registerEnokiWallets>[0], 'client'>;
+	useSuiClientContext: () => { client: SuiClient; network: string };
+	children: React.ReactNode;
+}) {
+	const { client, network } = useSuiClientContext();
+
+	const { wallets, unregister } = useMemo(
+		() => registerEnokiWallets({ ...config, client, network }),
```

Comment:
I think we want to do this as a side effect so A). the wallet registration code doesn't run during SSR since it's in the render code and makes calls to the `window` API and B). the effect becomes symmetric 

```
const [wallets, setWallets] = useState<Wallet[]>([]);

useEffect(() => {
  const unregister = registerEnokiWallets({ ...config, client, network });
  return unregister;
}, [config, client, network]);
```